x/application/keeper/msg_server_update_param.go

```diff
@@ -2,17 +2,67 @@ package keeper
 
 import (
 	"context"
+	"fmt"
 
-	sdk "github.com/cosmos/cosmos-sdk/types"
+	"google.golang.org/grpc/codes"
+	"google.golang.org/grpc/status"
 
-	"github.com/pokt-network/poktroll/x/application/types"
+	apptypes "github.com/pokt-network/poktroll/x/application/types"
 )
 
-func (k msgServer) UpdateParam(goCtx context.Context, msg *types.MsgUpdateParam) (*types.MsgUpdateParamResponse, error) {
-	ctx := sdk.UnwrapSDKContext(goCtx)
+func (k msgServer) UpdateParam(ctx context.Context, msg *apptypes.MsgUpdateParam) (*apptypes.MsgUpdateParamResponse, error) {
+	logger := k.logger.With(
+		"method", "UpdateParam",
+		"param_name", msg.Name,
+	)
 
-	// TODO: Handling the message
-	_ = ctx
+	if err := msg.ValidateBasic(); err != nil {
+		return nil, err
+	}
 
-	return &types.MsgUpdateParamResponse{}, nil
+	if k.GetAuthority() != msg.Authority {
+		return nil, status.Error(
+			codes.InvalidArgument,
+			apptypes.ErrAppInvalidSigner.Wrapf(
+				"invalid authority; expected %s, got %s",
+				k.GetAuthority(), msg.Authority,
+			).Error(),
+		)
+	}
+
+	params := k.GetParams(ctx)
+
+	switch msg.Name {
+	// TODO_IMPROVE: Add a Uint64 asType instead of using int64 for uint64 params.
```

Comment:
Would appreciate if you can tend to it in a short follow up PR?

You have the most context and it's cached in your mind.

Reply:
- #859